thrift0.14/src/main/java/com/linecorp/armeria/internal/common/thrift/TByteBufTransport.java

```diff
@@ -0,0 +1,28 @@
+package com.linecorp.armeria.internal.common.thrift;
+
+import org.apache.thrift.TConfiguration;
+import org.apache.thrift.transport.TTransportException;
+
+import io.netty.buffer.ByteBuf;
+
+public final class TByteBufTransport extends AbstractTByteBufTransport {
+
+    public TByteBufTransport(ByteBuf buf) {
+        super(buf);
+    }
+
+    @Override
+    public TConfiguration getConfiguration() {
+        return TConfiguration.DEFAULT;
+    }
+
+    @Override
+    public void updateKnownMessageSize(long size) throws TTransportException {
+        // This method is not called by the 'TProtocol's provided by Armeria
+    }
+
+    @Override
+    public void checkReadBytesAvailable(long numBytes) throws TTransportException {
+        // The size of readable bytes is already checked by the underlying ByteBuf.
```

Comment:
How about throwing `UnsupportedOperationException` for all three methods?

Reply:
This method is actually called by `libthrift` such as `TBinaryProtocol` before or after reading some bytes.
https://github.com/apache/thrift/blob/7e537505f014d8ede86f30a7500136c2f3e99505/lib/java/src/org/apache/thrift/protocol/TBinaryProtocol.java#L293-L298
I have no idea why Thrift checks the available read bytes after reading. 🤔

I guess `updateKnownMessageSize()` is not called by `libthrift`. I prefer to silently ignore like Thrift did in:
https://github.com/apache/thrift/blob/7e537505f014d8ede86f30a7500136c2f3e99505/lib/java/src/org/apache/thrift/transport/TFileTransport.java#L587-L595

Reply:
Ah I didn't know about it.
Then we should also return `null` for `getConfiguration()`?

Reply:
Yeah, there is no API that calls `getConfiguration()`.